src/azure-cli/azure/cli/command_modules/rdbms/_client_factory.py

```diff
@@ -54,19 +54,19 @@ def get_mysql_management_client(cli_ctx, **_):
     if rm_uri_override:
         client_id = getenv(CLIENT_ID)
         if client_id:
-            from azure.common.credentials import ServicePrincipalCredentials
-            credentials = ServicePrincipalCredentials(
+            from azure.identity import ClientSecretCredential
+            credentials = ClientSecretCredential(
                 client_id=client_id,
-                secret=getenv(CLIENT_SECRET),
-                tenant=getenv(TENANT_ID))
+                client_secret=getenv(CLIENT_SECRET),
+                tenant_id=getenv(TENANT_ID))
```

Comment:
Unless `AZURE_AUTHORITY_HOST` is specified, this doesn't work for sovereign clouds or Azure stacks, since authority of `cli_ctx` is not passed to `ClientSecretCredential`.